CPDShell/Core/algorithms/knn_algorithm.py

```diff
@@ -1,58 +1,65 @@
 """
-Module for implementation of CPD algorithm based on nearest neighbours.
+Module for implementation of CPD algorithm based on knn classification.
 """
 
 __author__ = "Artemii Patov"
 __copyright__ = "Copyright (c) 2024 Artemii Patov"
 __license__ = "SPDX-License-Identifier: MIT"
 
 import typing as tp
-from collections import deque
 from collections.abc import Iterable
-from math import sqrt
 
 import numpy as np
 
-import CPDShell.Core.algorithms.KNNCPD.knn_graph as knngraph
 from CPDShell.Core.algorithms.abstract_algorithm import Algorithm
+from CPDShell.Core.algorithms.ClassificationBasedCPD.abstracts.istatistic_test import TestStatistic
+from CPDShell.Core.algorithms.KNNCPD.knn_classifier import KNNClassifier
 
 
 class KNNAlgorithm(Algorithm):
     """
-    The class implementing change point detection algorithm based on nearest neighbours.
+    The class implementing change point detection algorithm based on classification.
     """
 
     def __init__(
         self,
-        metric: tp.Callable[[float, float], float] | tp.Callable[[np.float64, np.float64], float],
-        k=3,
-        threshold: float = 0.5,
+        distanceFunc: tp.Callable[[float, float], float] | tp.Callable[[np.float64, np.float64], float],
+        test_statistic: TestStatistic,
+        indent_coeff: float,
+        k=7,
```

Comment:
Maybe it'd be good to add in the class description info that this value was obtatined experimentally.